web3.js/src/util/send-and-confirm-transaction.ts

```diff
@@ -33,18 +34,36 @@ export async function sendAndConfirmTransaction(
     sendOptions,
   );
 
-  const status = (
-    await connection.confirmTransaction(
-      signature,
-      options && options.commitment,
-    )
-  ).value;
+  const status = await connection.getSignatureStatus(signature);
 
-  if (status.err) {
-    throw new Error(
-      `Transaction ${signature} failed (${JSON.stringify(status)})`,
+  const checkBlockHeight = async () => {
+    const blockHeight = await connection.getBlockHeight(
+      options && options.commitment,
     );
-  }
 
+    if (blockHeight > transaction.lastValidBlockHeight!) {
+      throw new Error('Transaction has expired.');
+    } else {
+      if (status) {
+        return;
+      }
+      await sleep(500);
```

Comment:
I wonder if we can use the knowledge of how far away we are from `lastValidBlockHeight` to minimize network chatter. Could you wait to poll this until we get close, then poll faster and faster as the last valid height approaches?